File: PYTHON/PROGRAM/Cpu.py
from Komponen import Komponen

class Cpu(Komponen):
    def __init__(self, jumlah_core=0, kecepatan_ghz=0.0, merk="", nama=""):
        super().__init__(merk, nama)
        self.__jumlah_core = jumlah_core
        self.__kecepatan_ghz = kecepatan_ghz
    
    # Getter
    def get_jumlah_core(self):
        return self.__jumlah_core
    
    def get_kecepatan_ghz(self):
        return self.__kecepatan_ghz
    
    # Setter
    def set_jumlah_core(self, jumlah_core):
        self.__jumlah_core = jumlah_core
    
    def set_kecepatan_ghz(self, kecepatan_ghz):
        self.__kecepatan_ghz = kecepatan_ghz

File: JAVA/PROGRAM/Cpu.java
public class Cpu extends Komponen {
    private int jumlahCore;
    private double kecepatanGhz;
    
    // Konstruktor
    public Cpu() {
        super();
        this.jumlahCore = 0;
        this.kecepatanGhz = 0.0;
    }
    
    public Cpu(int jumlahCore, double kecepatanGhz, String merk, String nama) {
        super(merk, nama);
        this.jumlahCore = jumlahCore;
        this.kecepatanGhz = kecepatanGhz;
    }
    
    // Getter dan Setter
    public int getJumlahCore() {
        return jumlahCore;
    }
    
    public void setJumlahCore(int jumlahCore) {
        this.jumlahCore = jumlahCore;
    }
    
    public double getKecepatanGhz() {
        return kecepatanGhz;
    }
    
    public void setKecepatanGhz(double kecepatanGhz) {
        this.kecepatanGhz = kecepatanGhz;
    }
}